qiskit_ibm_runtime/ibm_estimator.py

```diff
@@ -0,0 +1,147 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qiskit Runtime Estimator primitive service."""
+
+from typing import Iterable, Optional, Dict
+
+from qiskit.circuit import QuantumCircuit, Parameter
+from qiskit.quantum_info import SparsePauliOp
+
+from .base_primitive import BasePrimitive
+from .sessions.estimator_session import EstimatorSession
+
+
+class IBMEstimator(BasePrimitive):
```

Comment:
shouldn't this derive from BaseEstimator in terra as well?

Reply:
or is the idea to do that when that gets released in terra (and remove BasePrimitive here?)

Reply:
I think the idea is to remove after next terra release.

Reply:
Please take a look at the latest updates. We now have the four main classes:
```
IBMEstimator(BasePrimitive) - this is the factory class
Estimator(BaseEstimator) - this is returned by the above factory and inherits BaseEstimator
IBMSampler(BasePrimitive) - this is the factory class
Sampler(BaseSampler) - this is returned by the above factory and inherits BaseSampler
```

1. If the `BasePrimitive` abstraction looks ugly I can remove it and duplicate that code in IBMEstimator and IBMSampler init.
2. FYI I've copied the terra base classes here and will remove them once terra code is released.

cc: @levbishop